matrix/Euler.hpp

```diff
@@ -19,34 +19,113 @@ class Dcm;
 template <typename Type>
 class Quaternion;
 
+/**
+ * Euler angles class
+ *
+ * This class describes the transformation from the body fixed frame
+ * to the inertial frame via 3-2-1 tait brian euler angles.
```

Comment:
@priseborough agreed on intrinsic rotations since this is most convenient for UAVs